src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -677,17 +677,26 @@ protected void SelectVersion()
                     // Select the installed version by default.
                     // Otherwise, select the first version in the version list.
                     IEnumerable<DisplayVersion> possibleVersions = _versions.Where(v => v != null);
-                    SelectedVersion =
-                        possibleVersions.FirstOrDefault(v => v.Version.Equals(_searchResultPackage?.AllowedVersions?.OriginalString))
-                        ?? possibleVersions.FirstOrDefault(v => v.IsValidVersion);
+                    if (_filter.Equals(ItemFilter.UpdatesAvailable) || _filter.Equals(ItemFilter.All))
+                    {
+                        // If the latest version is not the installed version, we need to
+                        SelectedVersion = possibleVersions.FirstOrDefault(v => v.Range.OriginalString.Equals(latestVersion.ToString(), StringComparison.OrdinalIgnoreCase));
+                        UserInput = SelectedVersion.ToString();
+                    }
+                    else
+                    {
+                        SelectedVersion =
+                            possibleVersions.FirstOrDefault(v => StringComparer.OrdinalIgnoreCase.Equals(v.Range?.OriginalString, _searchResultPackage?.AllowedVersions?.OriginalString))
+                            ?? possibleVersions.FirstOrDefault(v => v.IsValidVersion);
+                        // Set the text of the combobox
+                        UserInput = _searchResultPackage.AllowedVersions?.OriginalString ?? SelectedVersion.ToString();
+                    }
```

Comment:
For the tabs Installed and Update we need to show the latest available version on the list, I preferred to pass the latestVersion as a parameter so we can set it faster.